src/utils/FormattingUtils.ts

```diff
@@ -47,10 +52,28 @@ export const formatFeePct = (relayerFeePct: BigNumber): string => {
  * Shortens a lengthy hexadecimal string to a shorter version with an ellipsis in the middle.
  * @param hex A hexadecimal string to be shortened.
  * @returns The shortened hexadecimal string.
- * @example shortenHexString("0x772871a444c6e4e9903d8533a5a13101b74037158123e6709470f0afbf6e7d94") -> "0x7787...7d94"
+ * @example createShortHexString("0x772871a444c6e4e9903d8533a5a13101b74037158123e6709470f0afbf6e7d94") -> "0x7787...7d94"
  */
-export function createShortHexString(hex: string): string {
-  return hex.substring(0, 5) + "..." + hex.substring(hex.length - 6, hex.length);
+export function createShortHexString(hex: string, maxLength = 8) {
+  // Store if the hex string has a leading 0x
+  const hasPrefix = /^0x/.test(hex);
+  // Remove the leading 0x if it exists
+  hex = hex.replace(/^0x/, "");
+  // If we have more maxLength then the hex size, we can simply
+  // return the hex directly.
+  if (hex.length <= maxLength) {
+    return hex;
+  }
+  // We can simulate rounding by adding 0.5 to the integer. If
+  // we had an odd division, the floor will add one additional
+  // character to the left side.
+  const leftCharacters = Math.floor(maxLength / 2 + 0.5);
+  // A simple floor division between the max character length
+  const rightCharacters = Math.floor(maxLength / 2);
+  // Combine the two sides.
+  const result = `${hex.substring(0, leftCharacters)}...${hex.substring(hex.length - rightCharacters)}`;
+  // Append the prefix if it was originally provided
+  return (hasPrefix ? "0x" : "") + result;
```

Comment:
As a user of this function, I'd intuitively expect that `maxLength` limits the output length of the resulting string - i.e. including including the `...`separator and optional `0x` prefix. If it does this, then there'd actually be no need to detect and strip off any leading `0x`. I would concede that the name `createShortHexString` seems a bit off though, since I'm describing something that works at the string level, and is basically ignorant of hexadecimal notation...

Reply:
Happy to dive into this if you feel there's anything actionable 🙂

Reply:
I do think we should be enforcing that the output string is no longer than `maxLength` characters, so the resulting string should be accounting for "0x" (if present) and "..." in the middle.

In general I think it'd be simpler if the code was just ignorant of whether the 0x prefix was there or not.

Reply:
Good call - resolved